src/object/player.hpp

```diff
@@ -552,6 +555,10 @@ class Player final : public MovingObject
 
   SurfacePtr m_airarrow; /**< arrow indicating Tux' position when he's above the camera */
 
+  std::array<SurfacePtr, 4> m_bubble_particles; /**< bubble particles for swimming */
+  Timer m_bubble_timer; /**< timer for spawning bubble particles */
+  std::list<std::pair<SurfacePtr, glm::vec2>> m_active_bubbles; /**< active bubble particles */
```

Comment:
Here you should use a vector.
```suggestion
  std::vector<std::pair<SurfacePtr, glm::vec2>> m_active_bubbles; /**< active bubble particles */
```
Don't forget to include `<vector>`!

Reply:
I suggested to not use std::vector here, because when bubbles are removed from m_active_bubbles, this may lead to unnecessary coping of the remaining elements. In my opinion a std::list is fine as it does not need the containing elements to be placed linearly on the heap.

Reply:
fair. Thanks for pointing that out.